TargetComposition/SymBuilder/SymBuilderTests/Program.cs

```diff
@@ -0,0 +1,447 @@
+        {
+            EngineOptions opts = new EngineOptions();
+            var arch = System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture;
+            switch(arch)
+            {
+                case System.Runtime.InteropServices.Architecture.X86:
+                    opts.EngineBitness = EngineArchitecture.X86;
+                    break;
+                case System.Runtime.InteropServices.Architecture.X64:
+                    opts.EngineBitness = EngineArchitecture.X64;
+                    break;
+                case System.Runtime.InteropServices.Architecture.Arm:
+                    opts.EngineBitness = EngineArchitecture.ARM32;
+                    break;
+                case System.Runtime.InteropServices.Architecture.Arm64:
+                    opts.EngineBitness = EngineArchitecture.ARM64;
+                    break;
+            }
+
+            DebugEngine engine = new DebugEngine();
+            engine.DmlOutput += EngineOutput;
+
+            int commandsProcessed = await SetupEngineForScript(engine, opts);
+            if (commandsProcessed == 0)
+            {
+                Console.WriteLine("    Unable to setup script: no harness commands recognized!");
+                return;
+            }
+
+            await engine.SendRequestAsync(new ExecuteRequest(".extpath+ " + Directory.GetCurrentDirectory()));
+            await engine.SendRequestAsync(new ExecuteRequest(".load SymbolBuilderComposition.dll"));
+
+            //
+            // It's important to note that DbgX does *NOT* have JavaScript extensibility by default.
+            // We need to explicitly load it from some debugger install.
+            //
+            string jsProviderPath = Path.Combine(m_debuggerInstallPath, "winext\\JsProvider.dll");
+            await engine.SendRequestAsync(new ExecuteRequest(".load " + jsProviderPath));
+
+            ClearOutput();
+            await engine.SendRequestAsync(new ExecuteRequest(".scriptload " + m_scriptPath));
+            loadOutput = m_lastOutput;
+            ClearOutput();
+
+            await engine.SendRequestAsync(new ExecuteRequest("dx @$testInit = @$scriptContents.initializeTests()"));
+
+            //
+            // The execution of initializeTests() should return a JS array of objects:
+            //
+            //     { Name: <test name>, Code: <test code> }
+            //
+            // While the code cannot be serialized across the wire, we can go back and manually ask the data model
+            // to invoke the test code and then deal with the results.
+            //
+            var testInfraXml = await engine.SendRequestAsync(new ModelQueryRequest("@$testInit,100000", false, DbgX.Interfaces.Enums.ModelQueryFlags.Default, recursionDepth: 2));
+            XDocument modelDoc = XDocument.Parse(testInfraXml);
+
+            //
+            // rootElement: The array itself...
+            //
+            XElement? rootElement = modelDoc.Root?.Element("Element");

```

Comment:
>XElement? rootElement = modelDoc.Root?.Element("Element") [](http://example.com/codeflow?start=12&length=57)

Maybe I just missed this in DbgX.  Is the whole "convert to dynamic" infrastructure available there...?

Reply:
No, that's in a higher layer right now (DbgX.Debugger). It's pretty self-contained, so I could move it to the DbgX layer at some point.

Reply:
Aah...  okay...  I'll leave the XML parsing for now then...  fortunately, most of this is done on the JS side, so there's a limited amount required in the harness...